src/steamship_langchain/llms/openai.py

```diff
@@ -1,63 +1,106 @@
 import hashlib
+import json
 import logging
-from typing import Any, List, Mapping, Optional
+from collections import defaultdict
+from typing import Any, Dict, Generator, List, Mapping, Optional
 
-import tiktoken
-from langchain.llms.base import BaseLLM, Generation, LLMResult
-from pydantic import Field
+from langchain.llms.base import Generation, LLMResult
+from langchain.llms.openai import BaseOpenAI
+from pydantic import root_validator
 from steamship import Block, File, Steamship, SteamshipError
 from steamship.data import TagKind, TagValueKey
 
-
-class OpenAI(BaseLLM):
+PLUGIN_HANDLE: str = "openai"
+ARGUMENT_WHITELIST = {
+    "client",
+    "model_name",
+    "temperature",
+    "max_tokens",
+    "top_p",
+    "frequency_penalty",
+    "presence_penalty",
+    "n",
+    "best_of",
+    "model_kwargs",
+    "openai_api_key",
+    "batch_size",
+    "request_timeout",
+    # "logit_bias",
+    "max_retries",
+    "callback_manager",
+    "cache",
+    "verbose",
+}
+
+
+class OpenAI(BaseOpenAI):
```

Comment:
+1. given the direction here, starting from BaseOpenAI seems like the right place.